Meta/Lagom/CMakeLists.txt

```diff
@@ -177,6 +178,33 @@ function(lagom_lib library fs_name)
         add_library(Lagom::${library} ALIAS ${target_name})
     endif()
 
+    # FIXME: do this properly and not here.
+    # Currently LibGUI is the only one that uses GML and we have no proper
+    # way to handle adding custom target dependenices to libraries.
+    if ("${library}" STREQUAL "GUI")
+        # Custom compile_gml function based on code_generators.cmake
+        function(compile_gml source output string_name)
```

Comment:
Instead of copying this whole function here, can we just add optional params to the function in code_generators.cmake to tweak its behavior for what we need here?